Release 1.3 (#265)
okteto/docs: src/content/self-hosted/install/releases.mdx

id: releases
---

## 1.3.0

### New Feature

> - Git Catalog feature will be part of the core of Okteto

Reviewer: ```suggestion
- Git Catalog to provide predefined development environments to every developer
```

Reviewer: Can we add an item about overwriting the autoscaler and init container images, and link the docs from here. This is for the case of companies that cannot pull from dockerhub. They can push `busybox` to their private registry, and overwrite these images

Reviewer: I would also mention https://github.com/okteto/app/pull/5045 and https://github.com/okteto/app/pull/5044 as features

Author: @pchico83 I found strange that I couldn't find the prs related to the autoscaler and init container in the changelog. Turns out they have not made it to the release 1.3

Here are the 2 prs:
https://github.com/okteto/app/pull/5224
https://github.com/okteto/app/pull/5223

Should we backport them?

Reviewer: @AgustinRamiroDiaz no, let's not add them to list of features then

Author: @pchico83 I'll remove the configuration associated to that too

### Bug Fix

- Use case insensitive check for GitHub ID

> - GitHub redeploy was having issues with the `repo` field in some cases

Reviewer: Can we be more specific about "having issues" and "in some cases"?

- Support for "resources" for the okteto daemonset

> - Fix blank namespaces

Reviewer: What's a blank namespace? can we elaborate more on which scenario is fixed?

> - Fix selecting a repository in the redeploy dialog

Reviewer: What is this fix? what was broken?

> - Show updated time instead of creation time in git deploys

Reviewer: ` git deploys` is an internal thing. we should say something like `Fix bug in the Okteto UI for the updated time of development environments`

id: releases
---

## 1.3.0

> ### New Feature

Reviewer 2: ```suggestion
### Features
```

> - Git Catalog to provide predefined development environments to every developer

Reviewer 2: Are we planning any documentation for this? (how to enable, how to use, etc...?). If not, I can try and take a stab at it this afternoon.

Reviewer 3: @viddo raised this same question and I'm happy to take a first pass. Would I just open a PR in the docs repo or do we have a different workflow for that?

Reviewer 4: You can open a PR directly if you want and comment it there. Or you can create a Notion doc and we can convert it easily later to include it in the docs. As far as I know we don't have a defined workflow for that...  do we? cc @RinkiyaKeDad

Reviewer 3: [Here's my first draft](https://www.notion.so/okteto/DRAFT-Git-Catalog-Documentation-0ac65fce66404410bf70033734cb7287)

> - Add Git Catalog tab to Okteto UI deploy

Reviewer 2: These two are the same right? I think it's better to put it on a single item. Also @omnomagonz @pchico83   could you help with a more user-friendly description? This is a powerful feature, I think we should mention a bit more why you should use this and how it benefits users.

Reviewer 3: My attempt at more user-friendly wording:

> Git Catalog that enables admins to pre-define a list of development environments that developers can use to deploy an environment without configuring it themselves

Something like that? I think we can (and should) expand further in the docs we create for this

Reviewer 5: > These two are the same right? …

Yes.

> … I think we can (and should) expand further in the docs we create for this

+1 - I guess as a general rule: release notes for the _what_, link to docs for the _how_ (to use it)? :)

Reviewer 3: > release notes for the what, link to docs for the how (to use it)?

For now, yep! In the future, I think it'd be great to do something like what we did at GitLab where the "release notes" were basically a blog post. For each major feature we explained:

* What is it?
* Why did we build it?
* Which customers (pricing tiers) get access to it
* A screenshot showing the feature
* A link to the epic or issue for it (if this becomes public later)

The minor features (and non-customer facing ones) were a bulleted list at the bottom.

All that to say: I'd love to eventually see the "what" _and_ "why" in the release notes, but a more thorough documentation or explanation of the feature done elsewhere. In the future anyway :)

Reviewer 2: @omnomagonz this is great, could you create a notion doc with these questions (maybe an entry inproduct on release notes guidance) so they don't get lost after we close this PR?

Reviewer 3: @rberrelleza you bet! [Here you go](https://www.notion.so/okteto/Monthly-Release-Notes-2afbe7e3e3c04c48aa99075331c07665)

> - Git Catalog to provide predefined development environments to every developer
> - Add Git Catalog tab to Okteto UI deploy
> - Remove CLI tab from Okteto UI deploy

Reviewer 5: @rlamana @omnomagonz Could we consolidate these to one entry that captures the gist of the new feature? Don't think we need this itemized breakdown, do we?

E.g. 
```suggestion
- Git Catalog, enabling predefined development environments
```

Also, do we have plans to add new documentation for how-to use enable and use the feature?

Reviewer 5: Doh, I obviously missed the other already existing thread 🤦 I guess I missed it because it was now "outdated"…
Resolving this thread, let's continue on the other one: https://github.com/okteto/docs/pull/265#discussion_r1051033934

> - Remove CLI tab from Okteto UI deploy

Reviewer 6: ```suggestion
- Remove CLI tab from Okteto UI deploy
```

Do you mean the "Deploy" section of the UI? If yes I think it would be better to frame it as: "Remove CLI tab from the Deploy section of Okteto UI" or maybe just leave it at: "Remove CLI tab from Okteto UI"

Reviewer 4: is this something we necessarily have to include in the notes?

Reviewer 2: Yeah, otherwise people might think that it's a bug or a misconfiguration. What's your concern @rlamana ?

Reviewer 4: No concern, just wondering. It makes sense, like you said it's a way to notify that something went away for a reason. I was just thinking about the "what" and "why", as the discussed in other comments, and wondering how to explain this one. I guess it would be: "to simplify and offer a better and seamless deploy experience in the UI".

### Bug Fix

> - Use case insensitive check for GitHub ID

Reviewer 2: ```suggestion
-  Verification of GitHub usernames is now case insensitive to match GitHub rules
```

> - GitHub redeploy modal was allowing `repo` field to be modified if opened fast enough

Reviewer 2: ```suggestion
- Fix the race condition that enabled users to modify `repo` field on the GitHub redeploy dialog
```

> - Support for "resources" for the okteto daemonset

Reviewer 2: This should be in the Feature lists. Can we also add a link to the configuration doc for this?

```suggestion
- You can now define  "resources" for the okteto daemonset component. Use this to ensure that Kubernetes will reserve resources for the `daemonset`.
```

> - Fix race condition that made namespaces in the Okteto UI empty  

Reviewer 2: ```suggestion
- Fix the race condition that prevented namespace-level resources from being displayed in the UI.
```

> - Fix bug in the Okteto UI for the updated time of development environments

Reviewer 2: What is the behavior being fixed here?

```suggestion
- Fix bug in the Okteto UI for the updated time of development environments
```

Reviewer 5: @rberrelleza it was using the _creation_ date instead of the _last-updated_ date, so wrong value displayed to users

So how about:
```suggestion
- Fix wrong "Last updated" time for development environments in the Okteto UI
```

Reviewer 2: ah I get it, looks good!

## 1.2.0

> ### New Feature

Reviewer 2: Unrelated to 1.3, but could you change this as well?

```suggestion
### Features
```
